### PR TITLE
Update setup.py from requirements

### PR DESCRIPTION
**Pull request by diego-plan9**
Update `setup.py` following previous changes to `requirements.txt`:
* remove `matplotlib` (already in extras)
* move `nxpd` to extras
* move `ipywidgets` to extras
* bump `sympy` to 1.3+
* reorder

### CODE REVIEW

**Comment by diego-plan9** [comment]
Follow-up of #1509 and #1515.

**Comment by diego-plan9** [comment]
> Ipywidgets is 7.3.0+ and pydot needs to be there as well as it is not installed with nxpd

Those changes are based on #1509, which does not include the quoted changes in `requirements-dev.txt` and ... was actually approved by you!

Can you double check and let me know if the new version that you quote in your comment are indeed the final ones, in order to fix `requirements-dev.txt` as well if is is the case?

**Comment by nonhermitian** [comment]
Yea, I was told the version was updated but did not check. I confirmed that the necessary GridBox is in 7.3.0+. Also the pydot is needed along with a separate binary installation of Graphviz for nxpd to work. The later of course is beyond our control.

**Comment by diego-plan9** [comment]
Thanks for the clarification, @nonhermitian ! Addressed it in bc8cd097a98d0cc9a3a9d667c6c9803c332cdfda

File: requirements.txt [hunk 1]
```diff
@@ -3,10 +3,10 @@ marshmallow>=2.16.3,<3
 marshmallow_polyfield>=3.2,<4
 networkx>=2.2
 numpy>=1.13
+pillow>=4.2.1
```
[the remark this answers is not in the document]
Comment (diego-plan9) [reply requirements.txt]: Wider question -  see https://github.com/Qiskit/qiskit-terra/issues/1517#issuecomment-447852716, as this PR is aimed for keeping the dependencies in sync.

**Comment by diego-plan9** [comment]
For the alphabetical part, the list of requirements grew "organically" - a bit of an order feels better than no order at all to my eyes, as we are not hitting the scenario you mention (yet).

**Pull request merged**